chia/wallet/wallet_node.py

```diff
@@ -626,21 +628,28 @@ async def receive_and_validate(inner_states: List[CoinState], inner_idx_start: i
                     if len(valid_states) > 0:
                         self.log.info(
                             f"new coin state received ({inner_idx_start}-"
-                            f"{inner_idx_start + len(inner_states) - 1}/ {len(items) + 1})"
+                            f"{inner_idx_start + len(inner_states) - 1}/ {len(items)})"
                         )
                         assert self.new_state_lock is not None
                         async with self.new_state_lock:
+                            if self.wallet_state_manager is None:
+                                return
                             await self.wallet_state_manager.new_coin_state(valid_states, peer, fork_height)
-                        if update_finished_height:
-                            await self.wallet_state_manager.blockchain.set_finished_sync_up_to(
-                                last_change_height_cs(inner_states[-1])
-                            )
+
+                            if update_finished_height:
+                                if len(cs_heights) == 1:
+                                    # We have processed all past tasks, so we can increase the height safely
+                                    synced_up_to = last_change_height_cs(valid_states[-1]) - 1
```

Comment:
If this is the last batch to finish, but not the highest batch, then we will end up setting the synced up to height below where we have actually synced to.  This is safe in the sense that it doesn't make you miss coins, it just makes you reprocess them.